docs/developer-guide/partition-data.rst

```diff
@@ -0,0 +1,111 @@
+sides of the join, are delivered to the same stream task during processing.
+If your input keys aren't co-partitioned, you need to re-key one of the inputs
+by using the PARTITION BY clause.
+
+Co-partitioning Requirements for Joins
+**************************************
+
+These are the requirements for data co-partitioning in a join:
+
+* The input data for the join, on the left and right side, must have the :ref:`same keying scheme <keys-have-same-keying-scheme>`.
+* The input data for the join must have the :ref:`same number of partitions <keys-have-same-number-of-partitions>`
+  on both sides.
+* The key for the join must be in the :ref:`same partition <keys-are-in-the-same-partition>` on both sides. 
+* Both sides of the join must have the :ref:`same partitioning strategy <keys-have-partitioning-strategy>`.  
+
+.. _keys-have-same-keying-scheme:
+
+Same Keying Scheme
+##################
+
+The input data for the join must have the same keying scheme, which means
+that the join must use the same key field on both sides.
+
+For example, you can join a stream of user clicks that's keyed by a ``userId VARCHAR``
+field with a table of user profiles that's keyed by a ``userId VARCHAR`` field. 
+The join won't match if the key fields don't have the same name and type.
+
+.. _keys-have-same-number-of-partitions:
+
+Number of Partitions is Equal
+#############################
+
+The input data for the join must have the same number of partitions on both
+sides.
+
+KSQL verifies this part of the co-partitioning requirement by ensuring an
+equal number of partitions for both sides of a join, otherwise KSQL throws
+a runtime exception.
+
+Use the ``<path-to-confluent>/bin/kafka-topics`` CLI tool
+with the ``--describe`` option to see the number of partitions for the
+Kafka topics that corresponding with your streams and tables.
+
+.. _keys-are-in-the-same-partition:
+
+Keys Are In The Same Partition
+##############################
+
+The key for the join must be in the same partition on both sides.
+
+For example, in a stream-table join, if a ``userId`` key with the value ``alice123``
+is in Partition 1 for the stream, but ``alice123`` is in Partition 2 for the table,
+the join won't match, even though both sides are keyed by ``userId``.
+
+.. _keys-have-partitioning-strategy:
+
+Keys Have the Same Partitioning Strategy
+########################################
+
+Both sides of the join must have the same partitioning strategy, or keying
+scheme.
+
+All applications that write to the input streams must have the same partitioning
+strategy, so records with the same key are delivered to same partition number.
+This means that the keyspace of the input data must be distributed across
+partitions in the same way on both sides of the join.
+
+KSQL can't verify whether the partitioning strategy matches between
+the input data of a join, so you must ensure this. If you use the default

```

Comment:
would this last sentence (`If you use the default`) make more sense as the first sentence in the paragraph?